build_cmake/configs/base.cmake

```diff
@@ -0,0 +1,166 @@
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#  All rights reserved.
+#
+#  See the file LICENSE for redistribution information
+#
+
+cmake_minimum_required(VERSION 3.11.0)
+
+include(build_cmake/helpers.cmake)
+
+# WiredTiger-related configuration options
+
+config_choice(
+    WT_ARCH
+    "Target architecture for WiredTiger"
+    OPTIONS
+        "x86;WT_X86;"
+        "arm64;WT_ARM64;"
+        "ppc64;WT_PPC64;"
+        "zseries;WT_ZSERIES;"
+)
+
+config_choice(
+    WT_OS
+    "Target OS for WiredTiger"
+    OPTIONS
+        "darwin;WT_DARWIN;"
+        "windows;WT_WIN;"
+        "linux;WT_LINUX;"
+)
+
+config_bool(
+    WT_POSIX
+    "Is a posix platform"
+    DEFAULT ON
+    DEPENDS "WT_LINUX OR WT_DARWIN"
+)
+
+config_string(
+    WT_BUFFER_ALIGNMENT_DEFAULT
+    "WiredTiger buffer boundary aligment"
+    DEFAULT 0
+)
+
+config_bool(
+    HAVE_DIAGNOSTIC
+    "Enable WiredTiger diagnostics"
+    DEFAULT OFF
+)
+
+config_bool(
+    HAVE_ATTACH
+    "Enable to pause for debugger attach on failure"
+    DEFAULT OFF
+)
+
+config_bool(
+    ENABLE_STATIC
+    "Compile as a static library"
+    DEFAULT OFF
+)
+
+config_bool(
+    ENABLE_STRICT
+    "Compile with strict compiler warnings enabled"
+    DEFAULT ON
+)
+
+config_bool(
+    ENABLE_PYTHON
+    "Configure the python API"
+    DEFAULT OFF
+    DEPENDS "NOT ENABLE_STATIC"
+)
+
+config_bool(
+    WT_STANDALONE_BUILD
+    "Support standalone build"
+    DEFAULT ON
+)
+
+config_bool(
+    HAVE_NO_CRC32_HARDWARE
+    "Disable any crc32 hardware support"
+    DEFAULT OFF
+)
+
+config_choice(
+    SPINLOCK_TYPE
+    "Set a spinlock type"
+    OPTIONS
+        "pthread;SPINLOCK_PTHREAD_MUTEX;"
+        "gcc;SPINLOCK_GCC;"
+        "msvc;SPINLOCK_MSVC;WT_WIN"
+        "pthread_adaptive;SPINLOCK_PTHREAD_ADAPTIVE;"
+)
+
+config_bool(
+    ENABLE_LIBLZ4
+    "Build the lz4 compressor extension"
+    DEFAULT OFF
+)
+
+config_bool(
+    ENABLE_LIBSNAPPY
+    "Build the snappy compressor extension"
+    DEFAULT OFF
+)
+
+config_bool(
+    ENABLE_LIBZ
```

Comment:
Typo ("ENABLE_ZLIB")?

Reply:
Good point, this might seem pretty confusing! I was initially following the autoconf convention of `HAVE_LIBX`, (autoconf exports the config macro `HAVE_LIBZ`, which I also export in `auto.cmake`). I essentially mirrored this convention when defining the `ENABLE_X` options. 

On second thought this is probably unnecessary so I'll rename this config to `ENABLE_ZLIB` to match the original command line option

Reply:
Updated!